File: app/src/main/java/com/github/overpass/gather/model/usecase/confirm/ConfirmEmailUseCase.java
package com.github.overpass.gather.model.usecase.confirm;

import androidx.lifecycle.LiveData;

import com.github.overpass.gather.screen.auth.register.confirm.ConfirmEmailStatus;
import com.github.overpass.gather.model.repo.confirm.ConfirmEmailRepo;

public class ConfirmEmailUseCase {

    private final ConfirmEmailRepo confirmEmailRepo;

    public ConfirmEmailUseCase(ConfirmEmailRepo confirmEmailRepo) {
        this.confirmEmailRepo = confirmEmailRepo;
    }

    public LiveData<ConfirmEmailStatus> confirmEmail() {
        return confirmEmailRepo.confirmEmail();
    }
}

File: app/src/main/java/com/github/overpass/gather/model/usecase/confirm/ConfirmEmailUseCase.kt
package com.github.overpass.gather.model.usecase.confirm

import com.github.overpass.gather.model.repo.confirm.ConfirmEmailRepo
import com.google.android.gms.tasks.Task
import javax.inject.Inject

class ConfirmEmailUseCase @Inject constructor(private val confirmEmailRepo: ConfirmEmailRepo) {

    fun confirmEmail(): Task<Void> = confirmEmailRepo.confirmEmail()

    fun isEmailVerified(): Boolean = confirmEmailRepo.isEmailVerified()
}
